### PR TITLE
chore: bump python dependencies

### PR DESCRIPTION
**Pull request by villebro**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [x] Build / Development Environment
- [x] Documentation

### SUMMARY
To ensure we're not missing out on security/bug fixes and general improvements, this PR bumps core dependencies to their latest stable versions. The biggest change is brought about by bumping `werkzeug` from `0.16` to `1.0.1`, in which the `contrib.cache` module has been moved into its own PyPI package `cachelib`. The `sqlalchemy-utils` breaking change will need to be revisited later once the project decides on the appropriate path forward.

### TEST PLAN
Local testing + CI

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,89 +4,91 @@
...
+markupsafe==1.1.1         # via jinja2, mako, wtforms
 marshmallow-enum==1.5.1   # via flask-appbuilder
-marshmallow-sqlalchemy==0.21.0  # via flask-appbuilder
-marshmallow==2.19.5       # via flask-appbuilder, marshmallow-enum, marshmallow-sqlalchemy
-more-itertools==8.1.0     # via zipp
-msgpack==0.6.2            # via apache-superset (setup.py)
-numpy==1.18.1             # via pandas, pyarrow
+marshmallow-sqlalchemy==0.23.0  # via flask-appbuilder
+marshmallow==2.21.0       # via flask-appbuilder, marshmallow-enum, marshmallow-sqlalchemy
+msgpack==1.0.0            # via apache-superset (setup.py)
```
Comment (dpgaspar) [inline requirements.txt:57]: always a bit scared of `1.0.0` versions

**Review by dpgaspar — APPROVED** [review]

File: requirements.txt (hunk 1 above)
Comment (villebro) [reply requirements.txt:57]: The way I see it, moving from `0.x` to `1.x` is usually a good thing in itself in the long term, and unless unit tests are picking up errors, we should be able to assume there were no relevant breaking changes.

File: requirements-dev.txt [hunk 2]
```diff
@@ -15,22 +15,23 @@
 # limitations under the License.
 #
 black==19.10b0
+cachelib==0.1
```
Comment (john-bodley) [inline requirements-dev.txt:18]: Shouldn’t this be defined in `setup.py`?

Comment (villebro) [reply requirements-dev.txt:18]: You're probably right, I guess we should consider this an explicit requirement.

**Review by john-bodley — APPROVED** [review]

**Comment by etr2460** [comment]
@villebro, @bkyryliuk, it looks like this broke s3cache since s3werkzeugcache relies on the old werkzeug format. I'm going to work on a fix now, but I think i'll need help releasing a new version on pypi https://github.com/bkyryliuk/s3werkzeugcache

**Comment by Clockmaker** [comment]
Hi guys, updating to `werkzeug` `1.0.1`, seems to break the oauth because `SecurityManager` is using `flask-appbuilder` that relies on `flask-oauthlib` using the old werkzeug's imports:
````python
from werkzeug import url_quote, url_decode, url_encode
````

Right now, one workaround is to import `cachelib` and downgrading `werkzeug`.

Am I missing something?
I can also provide more logs if needed, but I should probably open a new issue, let me know.

**Comment by villebro** [comment]
Ping @dpgaspar

**Comment by timesking** [comment]
> s3cache

I got the same problem, may I ask how you work around it?

**Pull request merged**